commons/src/quote.rs

```diff
@@ -0,0 +1,248 @@
+use anchor_client::solana_sdk::pubkey::Pubkey;
+use anyhow::{Context, Result};
+use lb_clmm::{
+    state::{
+        bin::{BinArray, SwapResult},
+        bin_array_bitmap_extension::BinArrayBitmapExtension,
+        lb_pair::LbPair,
+    },
+    utils::pda::derive_bin_array_pda,
+};
+use std::{collections::HashMap, time::SystemTime};
+
+#[derive(Debug)]
+pub struct SwapQuote {
+    pub amount_out: u64,
+    pub fee: u64,
+}
+
+pub fn quote_exact_in(
+    lb_pair_pubkey: Pubkey,
+    lb_pair: &LbPair,
+    mut amount_in: u64,
+    swap_for_y: bool,
+    bin_arrays: HashMap<Pubkey, BinArray>,
+    bitmap_extension: Option<&BinArrayBitmapExtension>,
+) -> Result<SwapQuote> {
+    let current_timestamp = SystemTime::now()
```

Comment:
I think we should make it as input parameters, and this should be node time (not system time)
So if someone want to integrate on-chain they can use this function (by passing clocktime)

Reply:
Fixed